.circleci/config.yml

```
@@ -240,24 +267,27 @@ jobs:
       - image: cimg/node:16.15
     steps:
       - show-npm-version
-      - provision-and-package-solcjs
+      - attach_workspace:
+          at: workspace
       - provision-hardhat-with-packaged-solcjs
       - run:
           name: Run hardhat-core test suite with latest solc
           command: |
-            HARDHAT_TESTS_SOLC_PATH="${PWD}/solc-js/soljson.js"
-            HARDHAT_TESTS_SOLC_VERSION=$(jq --raw-output .version solc-js/package.json)
+            cd hardhat/
+            HARDHAT_TESTS_SOLC_PATH="${PWD}/node_modules/solc/soljson.js"
+            HARDHAT_TESTS_SOLC_VERSION=$(jq --raw-output .version node_modules/solc/package.json)
             export HARDHAT_TESTS_SOLC_PATH HARDHAT_TESTS_SOLC_VERSION
 
-            cd hardhat/packages/hardhat-core
+            cd packages/hardhat-core/
             yarn test
```

If we update the `cimg/node` version to 18+ we can use pnpm here, which is installed by default in that image but not for the `16.15`. Because I suspect this will start to fail with the latest hardhat update.

```suggestion
            pnpm test
```

---

Then we can change `provision-hardhat-with-packaged-solcjs` to use `pnpm` as `package-manager`. Adding `pnpm` as a `package-manager` option to the `inject-solc-js-tarball` command. What do you think?